src/providers/oauth2/mod.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  ManifestOAuth2ProviderSchema,
+  OAuth2ProviderOptions,
+} from "./types.ts";
+
+export const DefineOAuth2Provider = (
+  definition: OAuth2ProviderDefinitionArgs,
+): OAuth2Provider => {
+  return new OAuth2Provider(definition);
+};
+
+export enum OAuth2ProviderTypes {
+  CUSTOM = "CUSTOM",
```

Comment:
In the future, Slack will have some preconfigured providers such as Atlassian, Google, PagerDuty. This will be for developer ease so that developers don't have to specify some constants such as authorization, token, identity urls. In this case, this enum will have more providers like 👇 and developers can just use those.
```
export enum OAuth2ProviderTypes {
  CUSTOM = "CUSTOM",
  GOOGLE = "GOOGLE",
  ATLASSIAN = "ATLASSIAN",
  PAGERDUTY = "PAGERDUTY",
}
```